bundles/org.openhab.binding.ipcamera/src/main/java/org/openhab/binding/ipcamera/internal/DahuaHandler.java

```diff
@@ -212,42 +250,13 @@ public void channelRead(@Nullable ChannelHandlerContext ctx, @Nullable Object ms
         }
         try {
             String content = msg.toString();
-            if (content.startsWith("--myboundary") || content.startsWith("-- myboundary")) {
-                processEvent(content);
-                return;
-            }
             ipCameraHandler.logger.trace("HTTP Result back from camera is \t:{}:", content);
-            // determine if the motion detection is turned on or off.
-            if (content.contains("table.MotionDetect[0].Enable=true")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_MOTION_ALARM, OnOffType.ON);
-            } else if (content.contains("table.MotionDetect[" + nvrChannel + "].Enable=false")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_MOTION_ALARM, OnOffType.OFF);
-            }
-
-            // determine if the audio alarm is turned on or off.
-            if (content.contains("table.AudioDetect[0].MutationDetect=true")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_AUDIO_ALARM, OnOffType.ON);
-            } else if (content.contains("table.AudioDetect[0].MutationDetect=false")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_AUDIO_ALARM, OnOffType.OFF);
-            }
-
-            // Handle AudioMutationThreshold alarm
-            if (content.contains("table.AudioDetect[0].MutationThreold=")) {
-                String value = ipCameraHandler.returnValueFromString(content, "table.AudioDetect[0].MutationThreold=");
-                ipCameraHandler.setChannelState(CHANNEL_THRESHOLD_AUDIO_ALARM, PercentType.valueOf(value));
-            }
-
-            // CrossLineDetection alarm on/off
-            if (content.contains("table.VideoAnalyseRule[0][1].Enable=true")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_LINE_CROSSING_ALARM, OnOffType.ON);
-            } else if (content.contains("table.VideoAnalyseRule[0][1].Enable=false")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_LINE_CROSSING_ALARM, OnOffType.OFF);
-            }
-            // Privacy Mode on/off
-            if (content.contains("table.LeLensMask[0].Enable=true")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_PRIVACY_MODE, OnOffType.ON);
-            } else if (content.contains("table.LeLensMask[0].Enable=false")) {
-                ipCameraHandler.setChannelState(CHANNEL_ENABLE_PRIVACY_MODE, OnOffType.OFF);
+            String[] events = boundaryPattern.split(content);
+            if (events.length > 0) {
+                processSettings(events[0]);
+                for (int i = 1; i < events.length; i++) {
+                    processEvent(events[i]);
+                }
             }
```

Comment:
The suggested change will fail / cause an array out of bounds error if events.length == 0.

Reply:
When the pattern doesn't match (i.e. --myboundary is not found in the data), events.length will be 1, so my original code in the PR would work and would call processSettings, but the loop to call processEvent will not run because 1 < 1 is false, so it will work the same way as it was before the PR.

Reply:
NOTE: I edited the suggested change a few times as I had to check up how it worked. So please check the code that is showing now and not what got emailed through when I first posted it...

The issue I was trying to address with the suggestion, is that all replies from the camera through this code and not just ones from the event stream. Other GET calls to change settings or request the state of a cameras settings will not contain the --myboundary so if the boundary is found, it is a waste of time parsing it for settings changes.

If you think I am wrong still, then I can complile and test more tomorrow as this is a reflection on my java knowledge.

Reply:
Settings replies NEVER start with the boundary, however regarding if settings can ever arrive with either of those strings further inside the reply, I am not 100% sure because I am not an employee of Dahua. It is highly unlikely and never seen it. It would be very easy to fault find that has occurred as the settings changes, only occur when the binding first connects to a camera or a command/control is moved, making it easy to look at the TRACE logs to fault find. It would also only stop an openHAB control from matching the cameras state, which I see as minor. The gain from not parsing every event is worth doing IMHO so thank you for making the change.